### PR TITLE
Made submit and check_complete methods more robust

### PR DESCRIPTION
**Pull request by ndharms**
There was an issue where our rate of checking if a job was complete was too frequent and SLURM would frequently return errors looking like the following:

```
slurm_load_jobs error: Socket timed out on send/recv operation
```
or
```
sbatch: error: Batch job submission failed: Job violates accounting/QOS policy (job submit limit, user's size and/or time limits)
```
After speaking with some people at Research Computing at Northeastern, we were told that our code was running `squeue` and `sbatch` too frequently and that we can check every minute (at minimum). This sets the `time.sleep()` to 90s so we avoid this issue. We also added changes to make sure that jobs are submitting by waiting for the queue to get smaller so our jobs don't overwhelm it.

### CODE REVIEW

**Comment by ndharms** [comment]
Okay, I'm thinking this is okay to merge this in even though the patch wasn't successful. We only changed 15 seconds to 90 seconds so nothing should have changed. Any complaints?

**Comment by ndharms** [comment]
Okay, I'm gonna do a rebase to fix the history and then add some tests to `job.py` to account for the decrease in coverage, but then this should be good to go.

**Comment by ndharms** [comment]
Okay, this is looking like all the checks pass and the coverage is good. I’m going to do a rebase to fix up the git history. But it should be ready for review soon.

**Comment by ndharms** [comment]
Hi there @davidfarinajr and @rwest, can one of y'all give this a review when you get a chance? Thanks!

File: autotst/job/job.py [hunk 1]
```diff
@@ -397,8 +488,9 @@ def calculate_species(self, species):
                 for running in currently_running:
                     if not running.is_alive():
                         currently_running.remove(name)
-                time.sleep(15)
```
Comment (rwest) [inline autotst/job/job.py:400]: Is this indentation level change intentional?
Looks a bit like the while loop will now run indefinitely without delay.

**Comment by rwest** [comment]
It looks like [more changed](https://github.com/ReactionMechanismGenerator/AutoTST/pull/57/files) than is described in the pull request description. Problem with rebasing?

**Comment by ndharms** [comment]
@rwest sorry about that. This PR was originally just about a sleep issue but we broadened the scope of the PR after speaking with @davidfarinajr in person. The PR description now reflects these changes.

File: autotst/job/job.py (hunk 1 above)
Comment (ndharms) [reply autotst/job/job.py:400]: Good catch. That was an intentional change in indentation but we should have a `time.sleep` in this loop. The latest commit reflects that change.

**Pull request merged**